File: 04_Microbot/Python/controller.py
#robot-ctrl.py
#Robot controller

from microbit import *
import radio

radio_group = 0 #number of radio frequency to use
forward = False
group_set = False #has the radio group been set?
radio.on()
radio.config(power=7)

stop_sign = Image(  "09990:"
                    "90099:"
                    "90909:"
                    "99009:"
                    "09990")

while True:# do forever
        if group_set is False:
            display.show(str(radio_group))

        if button_a.is_pressed() and button_b.is_pressed():
            if group_set is False:
                group_set = True
                radio.config(channel = radio_group)
                display.show(Image.HAPPY)
                forward = False
            else:
                if forward == False:
                    radio.send("forward")
                    forward = True
                    display.show(Image.ARROW_N)
                else:
                    radio.send("stop")
                    forward = False
                    display.show(stop_sign)
            
        elif button_a.is_pressed():
            if group_set == False and radio_group > 0 :
                radio_group = radio_group - 1
            else:
                radio.send("left")
                display.show(Image.ARROW_W)
                
        elif button_b.is_pressed():
            if group_set is False and radio_group < 255:
                radio_group = radio_group + 1 
            else:
                radio.send("right")
                display.show(Image.ARROW_E)
        sleep(200)

File: 04_Microbot/Javascript/controller.js
let groupSet = 0
let radioGroup = 0
input.onButtonPressed(Button.A, () => {
    if (groupSet == 0 && radioGroup > 0) {
        radioGroup = radioGroup - 1
    } else {
        radio.sendString("left")
        basic.showLeds(`
            . . # . .
            . # . . .
            # # # # #
            . # . . .
            . . # . .
            `)
    }
})
input.onButtonPressed(Button.AB, () => {
    if (groupSet == 0) {
        groupSet = 1
        radio.setGroup(radioGroup)
        basic.showIcon(IconNames.Happy)
    } else {
        radio.sendString("forward")
        basic.showLeds(`
            . . # . .
            . # # # .
            # . # . #
            . . # . .
            . . # . .
            `)
    }
})
input.onButtonPressed(Button.B, () => {
    if (groupSet == 0 && radioGroup < 255) {
        radioGroup = radioGroup + 1
    } else {
        radio.sendString("right")
        basic.showLeds(`
            . . # . .
            . . . # .
            # # # # #
            . . . # .
            . . # . .
            `)
    }
})
radioGroup = 0
while (groupSet != 1) {
    basic.showNumber(radioGroup)
}
basic.showIcon(IconNames.Happy)
basic.pause(100)
basic.showNumber(radioGroup)
